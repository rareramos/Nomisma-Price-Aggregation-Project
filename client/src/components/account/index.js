export { Account as default} from './account-container';
