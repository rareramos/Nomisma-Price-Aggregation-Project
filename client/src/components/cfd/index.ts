export { Cfd } from './cfd-container';
