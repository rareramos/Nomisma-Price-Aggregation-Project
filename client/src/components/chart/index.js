export { Chart } from './chart-container';
