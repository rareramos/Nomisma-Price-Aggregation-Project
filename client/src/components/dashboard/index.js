export { Dashboard } from './dashboard-container';
