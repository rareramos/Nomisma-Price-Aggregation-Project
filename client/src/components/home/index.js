export { Home as default } from './home-container';
