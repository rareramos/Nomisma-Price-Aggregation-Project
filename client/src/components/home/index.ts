export { Home } from './home-container';
