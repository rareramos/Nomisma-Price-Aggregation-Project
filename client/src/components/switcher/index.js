export { Switcher } from './switcher-container';
