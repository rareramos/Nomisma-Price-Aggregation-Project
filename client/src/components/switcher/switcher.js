import React from 'react';
import {
  func, arrayOf, object, number,
} from 'prop-types';
import { SecondaryNav, TabsSecondaryItem } from '@nomisma/nomisma-ui/tabs/styles';
import { useTabs } from '@nomisma/nomisma-ui/tabs';
import { Styled } from './styled';

export const Switcher = ({ tabs, selectedSource, onClick }) => {
  const { getTabProps } = useTabs();
  return (
    <Styled>
      <SecondaryNav
        selected={selectedSource}
        setActiveTab={index => onClick(tabs[index].value)}
      >
        {
          tabs.map((tab, index) => (
            <TabsSecondaryItem
              {...getTabProps(index)}
            />
          ))
        }
      </SecondaryNav>
    </Styled>
  );
};

Switcher.propTypes = {
  selectedSource: number.isRequired,
  tabs: arrayOf(object).isRequired,
  onClick: func.isRequired,
};
