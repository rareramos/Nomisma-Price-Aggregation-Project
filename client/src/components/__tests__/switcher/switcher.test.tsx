import React from 'react';
import { shallow } from 'enzyme';

import { Switcher as SwitcherComp } from 'components/switcher/switcher';
import { ISwitcherProps } from 'types/components/switcher';
import { getTabs } from 'utils';

const SwitcherProps : ISwitcherProps = {
  onClick: jest.fn,
  selectedSource: 1,
  tabs: getTabs(),
};

describe('Switcher test', () => {
  it('matches switcher component snapshot', () => {
    const wrapper = shallow(<SwitcherComp {...SwitcherProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
