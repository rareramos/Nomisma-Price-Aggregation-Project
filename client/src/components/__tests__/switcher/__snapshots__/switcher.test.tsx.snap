// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher test matches switcher component snapshot 1`] = `
<styled.div>
  <styled.ul
    selected={1}
    setActiveTab={[Function]}
  >
    <styled.li
      key="0"
      onClick={[Function]}
      selected={true}
    />
    <styled.li
      key="1"
      onClick={[Function]}
      selected={false}
    />
    <styled.li
      key="2"
      onClick={[Function]}
      selected={false}
    />
  </styled.ul>
</styled.div>
`;
